packages/market-service/src/coingecko/coingecko.ts

```diff
@@ -38,6 +43,41 @@ const coingeckoIDMap: CoinGeckoIDMap = Object.freeze({
 export class CoinGeckoMarketService implements MarketService {
   baseUrl = 'https://api.coingecko.com/api/v3'
 
+  private readonly defaultGetByMarketCapArgs: GetByMarketCapArgs = {
+    pages: 10,
+    perPage: 250
+  }
+
+  async getByMarketCap(args?: GetByMarketCapArgs) {
+    const argsToUse = args ?? this.defaultGetByMarketCapArgs
```

Comment:
you should merge the two instead so they can provide just "pages" or 'perPage" and still get the default for the other, unless that doesn't make sense.

Reply:
done